core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts

```diff
@@ -1193,16 +1466,111 @@ export class TopicManager extends BaseUser {
     await this.expectElementToBeVisible(confirmMoveButton, false);
   }
 
+  /**
+   * Checks if the skill is visible in the skill selection modal.
+   * @param skillName The name of the skill.
+   * @param visible Whether the skill should be visible or not.
+   */
+  async expectSkillInSkillSelectionModalToBeVisible(
+    skillName: string,
+    visible: boolean = true
+  ): Promise<ElementHandle | null> {
+    const skillVisible = await this.isElementVisible(skillItem);
+    if (!skillVisible) {
+      if (visible) {
+        throw new Error(
+          `Skill ${skillName} is not visible in the skill selection modal.`
+        );
+      } else {
+        showMessage(
+          `Skill ${skillName} is not visible in the skill selection modal.`
+        );
+        return null;
+      }
+    }
+
+    const skillElements = await this.page.$$(skillItem);
+    for (const skillElement of skillElements) {
+      const foundSkillName = await this.page.evaluate(
+        (skillElement: Element) => skillElement.textContent?.trim(),
+        skillElement
+      );
+      if (skillName === foundSkillName) {
+        if (visible) {
+          return skillElement;
+        } else {
+          throw new Error(
+            `Skill ${skillName} is visible in the skill selection modal.`
+          );
+        }
+      }
+    }
+
+    if (visible) {
+      throw new Error(
+        `Skill ${skillName} is not visible in the skill selection modal.`
+      );
+    } else {
+      showMessage(
+        `Skill ${skillName} is visible in the skill selection modal.`
+      );
+      return null;
+    }
+  }
+
+  /**
+   * Fills the skill name input field with the given skill name.
+   * @param {string} skillName - The skill name to fill the input field with.
+   */
+  async fillSkillNameInSkillSelectionModal(skillName: string): Promise<void> {
+    await this.expectElementToBeVisible(skillNameInputSelector);
+    await this.type(skillNameInputSelector, skillName);
+    await this.expectElementValueToBe(skillNameInputSelector, skillName);
+  }
+
+  /**
+   * Selects the skill with the given name and clicks on the "Done" button in
+   * the Skill Selection Modal.
+   * @param {string} skillName - The name of the skill to select.
+   */
+  async selectSkillAndClickOnDoneInSkillSelectionModal(
+    skillName: string
+  ): Promise<void> {
+    const skillElement =
+      await this.expectSkillInSkillSelectionModalToBeVisible(skillName);
+
+    if (!skillElement) {
+      throw new Error(`Skill ${skillName} not found in Skill Selection Modal`);
+    }
+    const radioInnerCircleSelectorElement = await skillElement.waitForSelector(
+      radioInnerCircleSelector
+    );
+
+    if (!radioInnerCircleSelectorElement) {
+      throw new Error('Radio inner circle selector not found');
+    }
+
+    await this.page.evaluate(selector => {
+      document.querySelector(selector).click();
+    }, radioInnerCircleSelector);
```

Comment:
why can't we just use `this.clickOn()` here?

Reply:
I copied it from the other function.
Done.